defs/go.bzl

```diff
@@ -87,7 +92,7 @@ def _go_genrule_impl(ctx):
 _go_genrule = go_rule(
     _go_genrule_impl,
     attrs = {
-        "srcs": attr.label_list(allow_files = True),
+        "srcs": attr.label_list(allow_files = True, providers = [GoLibrary]),
```

Comment:
The way we're using `go_genrule` in kubernetes, we're treating it as a slightly more sophisticated wrapper around `native.genrule`, so `srcs` is just files, not go sources. Our go dependencies go in `go_deps`.

For example, see
https://github.com/kubernetes/kubernetes/blob/2fb9fc24001c54bcfbf2ef4840ff1e86867e2d70/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD#L25-L52,
where we use `srcs` for a boilerplate text file and `go_deps` for relevant go libraries.

Would it make more sense to pass `go_deps` through to the implementation and use that for the `importpath`s logic, rather than using `srcs`? (We'd still additionally use `go_deps` for the `go_paths` construction.)